src/View/Helper/FormHelper.php

```diff
@@ -1502,13 +1502,20 @@ protected function _magicOptions($fieldName, $options, $allowOverride)
             unset($options['step']);
         }
 
-        $autoLength = !array_key_exists('maxlength', $options)
-            && !empty($fieldDef['length'])
-            && $options['type'] !== 'select';
-
         $allowedTypes = ['text', 'textarea', 'email', 'tel', 'url', 'search'];
```

Comment:
Nitpick: Can you please rename this variable to a more descriptive `$typesWithMaxlength`.